### ISSUE TITLE
Null pointer dereference in mark_context_stack

### ISSUE THREAD

**Issue by clayton-shopify** [mruby/mruby#3549]
The following input demonstrates a crash:
```ruby
def to_str
ensure @@a + + -> {}
  b ""
end

class A < to_str
end
```
ASAN report:
```
==5758==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x000101b49dfd bp 0x7fff5dcfec90 sp 0x7fff5dcfea80 T0)
==5758==The signal is caused by a READ memory access.
==5758==Hint: address points to the zero page.
    #0 0x101b49dfc in mark_context_stack gc.c:554
    #1 0x101b493a6 in mark_context gc.c:571
    #2 0x101b48157 in root_scan_phase gc.c:874
    #3 0x101b47330 in incremental_gc gc.c:1081
    #4 0x101b435cc in incremental_gc_step gc.c:1122
    #5 0x101b427dc in mrb_incremental_gc gc.c:1166
    #6 0x101b42148 in mrb_obj_alloc gc.c:507
    #7 0x101ac5ff3 in ary_new_capa array.c:30
    #8 0x101ac5dd7 in mrb_ary_new_capa array.c:41
    #9 0x101b2a40e in mrb_vformat (mruby:x86_64+0x10006840e)
    #10 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #11 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #12 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #13 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #14 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #15 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #16 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #17 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #18 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #19 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #20 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #21 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #22 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #23 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #24 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #25 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #26 0x101ad2521 in join_ary array.c:1052
    #27 0x101ad0e32 in mrb_ary_join array.c:1079
    #28 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #29 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #30 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #31 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #32 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #33 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #34 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #35 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #36 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #37 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #38 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #39 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #40 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #41 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #42 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #43 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #44 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #45 0x101ad2521 in join_ary array.c:1052
    #46 0x101ad0e32 in mrb_ary_join array.c:1079
    #47 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #48 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #49 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #50 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #51 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #52 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #53 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #54 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #55 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #56 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #57 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #58 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #59 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #60 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #61 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #62 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #63 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #64 0x101ad2521 in join_ary array.c:1052
    #65 0x101ad0e32 in mrb_ary_join array.c:1079
    #66 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #67 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #68 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #69 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #70 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #71 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #72 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #73 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #74 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #75 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #76 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #77 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #78 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #79 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #80 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #81 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #82 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #83 0x101ad2521 in join_ary array.c:1052
    #84 0x101ad0e32 in mrb_ary_join array.c:1079
    #85 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #86 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #87 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #88 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #89 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #90 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #91 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #92 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #93 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #94 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #95 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #96 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #97 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #98 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #99 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #100 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #101 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #102 0x101ad2521 in join_ary array.c:1052
    #103 0x101ad0e32 in mrb_ary_join array.c:1079
    #104 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #105 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #106 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #107 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #108 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #109 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #110 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #111 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #112 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #113 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #114 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #115 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #116 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #117 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #118 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #119 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #120 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #121 0x101ad2521 in join_ary array.c:1052
    #122 0x101ad0e32 in mrb_ary_join array.c:1079
    #123 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #124 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #125 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #126 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #127 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #128 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #129 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #130 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #131 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #132 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #133 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #134 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #135 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #136 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #137 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #138 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #139 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #140 0x101ad2521 in join_ary array.c:1052
    #141 0x101ad0e32 in mrb_ary_join array.c:1079
    #142 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #143 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #144 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #145 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #146 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #147 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #148 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #149 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #150 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #151 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #152 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #153 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #154 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #155 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #156 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #157 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #158 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #159 0x101ad2521 in join_ary array.c:1052
    #160 0x101ad0e32 in mrb_ary_join array.c:1079
    #161 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #162 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #163 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #164 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #165 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #166 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #167 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #168 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #169 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #170 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #171 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #172 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #173 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #174 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #175 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #176 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #177 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #178 0x101ad2521 in join_ary array.c:1052
    #179 0x101ad0e32 in mrb_ary_join array.c:1079
    #180 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #181 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #182 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #183 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #184 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #185 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #186 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #187 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #188 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #189 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #190 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #191 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #192 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #193 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #194 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #195 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #196 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #197 0x101ad2521 in join_ary array.c:1052
    #198 0x101ad0e32 in mrb_ary_join array.c:1079
    #199 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #200 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #201 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #202 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #203 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #204 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #205 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #206 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #207 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #208 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #209 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #210 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #211 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #212 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #213 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #214 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #215 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #216 0x101ad2521 in join_ary array.c:1052
    #217 0x101ad0e32 in mrb_ary_join array.c:1079
    #218 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #219 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #220 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #221 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #222 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #223 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #224 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #225 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #226 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #227 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #228 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #229 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #230 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #231 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #232 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #233 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #234 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #235 0x101ad2521 in join_ary array.c:1052
    #236 0x101ad0e32 in mrb_ary_join array.c:1079
    #237 0x101b2b7d2 in mrb_vformat (mruby:x86_64+0x1000697d2)
    #238 0x101b2c1f1 in raise_va (mruby:x86_64+0x10006a1f1)
    #239 0x101b2c826 in mrb_name_error (mruby:x86_64+0x10006a826)
    #240 0x101c0f243 in mrb_mod_cv_get (mruby:x86_64+0x10014d243)
    #241 0x101c10732 in mrb_vm_cv_get (mruby:x86_64+0x10014e732)
    #242 0x101c2a3e9 in mrb_vm_exec (mruby:x86_64+0x1001683e9)
    #243 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #244 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #245 0x101c5601a in ecall (mruby:x86_64+0x10019401a)
    #246 0x101c2edbf in mrb_vm_exec (mruby:x86_64+0x10016cdbf)
    #247 0x101c26959 in mrb_vm_run (mruby:x86_64+0x100164959)
    #248 0x101c1f69e in mrb_run (mruby:x86_64+0x10015d69e)
    #249 0x101c1cfe7 in mrb_funcall_with_block (mruby:x86_64+0x10015afe7)
    #250 0x101c199b7 in mrb_funcall_argv (mruby:x86_64+0x1001579b7)
    #251 0x101b9f18e in convert_type (mruby:x86_64+0x1000dd18e)
    #252 0x101ba02aa in mrb_check_convert_type (mruby:x86_64+0x1000de2aa)
    #253 0x101bce12d in mrb_check_string_type (mruby:x86_64+0x10010c12d)
    #254 0x101ad2521 in join_ary array.c:1052
    #255 0x101ad0e32 in mrb_ary_join array.c:1079

==5758==Register values:
rax = 0x00007fff5dcfeaa0  rbx = 0x00007fff5dcfeae0  rcx = 0xbebebebebebebebe  rdx = 0x17d7d7d7d7d7d7d7
rdi = 0x00007fff5dcfeaa0  rsi = 0x0000100000000000  rbp = 0x00007fff5dcfec90  rsp = 0x00007fff5dcfea80
 r8 = 0xbebebebebebebe00   r9 = 0x000060b00000aa18  r10 = 0x00001c1600001543  r11 = 0x00001daf5dcf9160
r12 = 0x00007fff5dcff740  r13 = 0x00007fff5dcff760  r14 = 0x00007fff5dcff700  r15 = 0x00007fff5dcff720
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV gc.c:554 in mark_context_stack
==5758==ABORTING
Abort trap: 6
```
This issue was reported by https://hackerone.com/d4nny

**Comment by clayton-shopify** [comment]
https://hackerone.com/dgaletic reported the following input, which causes what appears to be the same crash:
```ruby
unless def to_str
    ensure E - {} rescue "".i "".dup.prepend "".class Time.new
    end
end
i i Struct.new "".lines
```

**Comment by clayton-shopify** [comment]
Another input from https://hackerone.com/dgaletic that produces the same crash (on Ubuntu+gcc only):
```ruby
class A < protected def to_str
    ensure
        super "".String 0
    end
    T
end
```

**Comment by matz** [comment]
All of above examples do not crash any longer. Probably fixed.

**Comment by clayton-shopify** [comment]
It appears these were addressed by https://github.com/mruby/mruby/commit/ab25eaea2345577b4e28672908f0190ce6b21684.

**Comment by clayton-shopify** [comment]
@matz https://hackerone.com/flamezzz noticed that the inputs in this issue are crashing again. I did a bisect and found that this started in e26e11860466676e5713fa0d9ab268ea8dbdcef9.

**Comment by dabroz** [comment]
Is there a reason we don't include these inputs in mruby regression tests?

**Comment by matz** [comment]
@clayton-shopify cef523cc fixed the issue (which is wrongly marked as #3637).

@dabroz it's a kind of difficult trade-off. If we add new regression tests for all issues, our test suites would become too big pretty soon and consume longer time to execute tests (reducing our productivity). Maybe we can create bigger test suites for regression checking.
